extended/src/main/java/apoc/ml/bedrock/Bedrock.java

```diff
@@ -0,0 +1,123 @@
+package apoc.ml.bedrock;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import apoc.Description;
+import apoc.result.ObjectResult;
+import apoc.util.ExtendedUtil;
+import apoc.util.Util;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import static apoc.ml.bedrock.BedrockInvokeConfig.MODEL_ID;
+import static apoc.util.JsonUtil.OBJECT_MAPPER;
+import static apoc.ml.bedrock.BedrockInvokeResult.*;
+import static apoc.ml.bedrock.BedrockUtil.*;
+
+
+public class Bedrock {
+    
+    @Procedure
+    public Stream<ModelItemResult> list(@Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
+
+        BedrockConfig conf = new BedrockGetModelsConfig(config);
+        
+        return executeRequestCommon(null, "modelSummaries[*]", conf)
+                .flatMap(i -> ((List<Map<String, Object>>) i).stream())
+                .map(ModelItemResult::new);
+    }
+    
+    @Procedure
+    @Description("To create a customizable bedrock call")
+    public Stream<ObjectResult> custom(@Name(value = "body") Object body,
```

Comment:
we should not use object bodies

either have something structured like a map that we we also document (see VertexAI)
or a string/list of strings for simple calls (like in the other ones)

model parameters, like top-k, top-p, temparature etc. should go into the config parameter and passed through

the main ones should also be documented, and also add links in the docs to AWS docs for the model(parameters)

Reply:
same for the other procs, but if we generalize it they should become two (embedding + chat-completion) or three (if you want to add image).

and additionally the custom one.